### PR TITLE
Include JWT on all portal requests (fix for auth-only portals)

### PR DESCRIPTION
**Pull request by dghelm**
# PULL REQUEST

## Overview

The `ky` requests to other subdomains on the portal were not including the JWT cookie, causing auth errors.

`ky` doesn't seem to expose fetch's `credentials` option, so I've replaced methods with `fetch` where needed.

### CODE REVIEW

File: src/services/getDappMetadata.js [hunk 1]
```diff
@@ -36,8 +35,12 @@ export default async function getDappMetadata(skylink) {
   const skynetMetadata = await getSkynetMetadata(skylink);
 
   try {
-    const skylinkUrl = await skynetClient.getSkylinkUrl(skylink, { subdomain: true });
-    const contentType = (await ky.head(skylinkUrl)).headers.get("content-type");
+    // grab skylinkUrl for use in fetching metadata.
+    // Doesn't use subdomain:true to ensure JWT cookie gets sent
+    const skylinkUrl = await skynetClient.getSkylinkUrl(skylink);
```
Comment (dghelm) [inline src/services/getDappMetadata.js:40]: I could maybe leave this as a subdomain actually, but open to discussion.

Comment (m-cat) [reply src/services/getDappMetadata.js:40]: I think the JWT should be valid for all subdomains actually. Should double-check with @kwypchlo . In that case we'd want to remove the comment about `subdomain:true`.

Comment (m-cat) [reply src/services/getDappMetadata.js:40]: Do we even need the credentials for getting metadata? Does fetching metadata count as a download in the user's account?

**Review by m-cat — CHANGES_REQUESTED** [review]
Couple comments. Also, can you please include instructions for how to test the deployed changes?

File: src/services/getDappMetadata.js (hunk 1 above)
Comment (dghelm) [reply src/services/getDappMetadata.js]: Removed the comment and restored subdomain:true in following commit.

We could probably use a different technique to get the content-type of the main file, but HEAD request by the browser is what it is currently.

**Comment by dghelm** [comment]
The app is usable at https://3004q31gj0ilrtpj40j8s2mq3st2mbgtgc3eq068qjs07at0cvmufgg.skynetpro.net/

Here is the current error for homescreen.hns.skynetpro.net for resolving an HNS name:
image

If able to resolve an HNS name (say, homescreen.hns because it's same-origin), you can get the metadata, but you can't call HEAD and then GET on the skylink:
image

**Review by m-cat — APPROVED** [review]
Cool, worked great in my testing. Thanks! 👍

**Comment by dghelm** [comment]
Following feedback on Discord, Karol preferred that we keep all calls using `ky`. Last 2 commits replace `ky-universal` dependency with `ky` (we don't need the nodejs context here) and return to the original code, but adding `{credentials: "include"}`

**Comment by dghelm** [comment]
> Deployed to https://3001covtrto3u7jbf394ieers01t5bskoph2lofr9c9s4nq7curmtu8.siasky.net/Skylink: `sia://GAAWY_3fcD8ea3jSSTnb4APSr5TGYirh-0sTwl9HZ7du-Q`

https://300buoi415mmk4mcu0mnsmoa6hk6s5bpcn6gbbppcng0j73iksolatg.skynetpro.net/

**Review by m-cat — APPROVED** [review]
Tested.

**Review by kwypchlo — APPROVED** [review]

**Pull request merged**